core/payment-driver/zksync/examples/withdrawal.rs

```diff
@@ -0,0 +1,111 @@
+#[macro_use]
+extern crate log;
+
+use bigdecimal::BigDecimal;
+use hex::ToHex;
+use structopt::StructOpt;
+use ya_zksync_driver::zksync::{faucet, utils};
+use zksync::zksync_types::{TxFeeTypes, H256};
+use zksync::{types::BlockStatus, Network, Provider, Wallet, WalletCredentials};
+use zksync_eth_signer::{EthereumSigner, PrivateKeySigner};
+
+const TOKEN: &str = "GNT";
+const PRIVATE_KEY: &str = "312776bb901c426cb62238db9015c100948534dea42f9fa1591eff4beb35cc13";
+
+#[derive(Clone, Debug, StructOpt)]
+struct Args {
+    #[structopt(long = "gen-key")]
+    genkey: bool,
+
+    #[structopt(long, default_value = "5.0")]
+    amount: f64,
+}
+
+#[actix_rt::main]
+async fn main() -> anyhow::Result<()> {
+    let log_level = std::env::var("RUST_LOG").unwrap_or("info".to_owned());
+    std::env::set_var("RUST_LOG", log_level);
+    env_logger::init();
+
+    let args: Args = Args::from_args();
+    let private_key = if args.genkey {
+        debug!("Using randomly generated key");
+        H256::random()
+    } else {
+        debug!("Using hardcoded key");
+        let mut bytes = [0u8; 32];
+        hex::decode_to_slice(PRIVATE_KEY, &mut bytes)
+            .expect("Cannot decode bytes from hex-encoded PK");
+        H256::from(bytes)
+    };
+
+    let pk_hex: String = private_key.encode_hex();
+    let signer = PrivateKeySigner::new(private_key);
+    let address = signer.get_address().await?;
+    let addr_hex = format!("0x{}", address.encode_hex::<String>());
+    info!("Private key: {}\nAccount address {}", pk_hex, addr_hex);
+
+    info!("Funding an account");
+    // let hex_addr = key.address().to_string();
+    faucet::request_ngnt(&addr_hex).await?;
+
+    info!("Creating wallet");
+    let provider = Provider::new(Network::Rinkeby);
+    let cred = WalletCredentials::from_eth_signer(address, signer, Network::Rinkeby).await?;
+    let wallet = Wallet::new(provider, cred).await?;
+
+    if wallet.is_signing_key_set().await? == false {
+        info!("Unlocking account");
+        let unlock = wallet
+            .start_change_pubkey()
+            .fee_token("GNT")?
+            .send()
+            .await?;
+        debug!("unlock={:?}", unlock);
+        unlock.wait_for_commit().await?;
+    }
+
+    let balance = wallet.get_balance(BlockStatus::Committed, TOKEN).await?;
+    info!(
+        "Deposit successful {} NGNT available",
+        utils::big_uint_to_big_dec(balance.clone())
+    );
+
+    info!("Obtaining withdrawal fee");
+    let amount = utils::big_dec_to_big_uint(BigDecimal::from(args.amount))?;
```

Comment:
its dangerous ( and limiting ) to use `f64` -> `BigDecimal`

I would recommend to use a string as input param ( for both `BigDecimal`'s and `big_int`'s to support higher precision

Reply:
Great catch! Fixed.